turbo/src/main/kotlin/dev/hotwire/turbo/util/TurboUriHelper.kt

```diff
@@ -104,6 +108,21 @@ internal class TurboUriHelper(val context: Context) {
             fileSize = 0
         )
     }
+
+    /**
+     * Checks for path traversal vulnerability (caused e.g. by input filename containing "../")
+     * which could lead to modification of the destination directory.
+     *
+     * More information: https://developer.android.com/topic/security/risks/path-traversal
+     */
+    private fun File.checkForPathTraversalVulnerability(destDirectory: File) {
+        val destinationDirectoryPath = destDirectory.canonicalPath
+        val outputFilePath = this.canonicalPath
```

Comment:
Just want to note that the `.canonicalPath` use can throw its own exceptions, but not an issue for the current implementation, since they'll be caught/handled in the caller above.

Reply:
I can also change the `checkForPathTraversalVulnerability` to return a `Boolean` instead of throwing an exception if you prefer 👍

Reply:
I lean slightly towards a boolean implementation so the exceptions are isolated and it mirrors the `File.originIsAppResource()` function, which is also for preventing vulnerabilities.

Reply:
Looks great, thanks!